compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2105,7 +2105,23 @@ impl CheckAttrVisitor<'_> {
     }
 
     fn check_macro_export(&self, hir_id: HirId, attr: &Attribute, target: Target) {
-        if target != Target::MacroDef {
+        // special case when `#[macro_export]` is applied to a macro 2.0
+        if target == Target::MacroDef {
```

Comment:
Should this be put as an extra `else` branch below?

Reply:
sure, that's fine with me - I wanted to keep the "`MacroDef` checks" grouped together but having it the else is also fine. I'll change it

Reply:
Addressed in 327361e187b